src/main/java/org/springframework/cloud/scheduler/spi/cloudfoundry/CloudFoundryAppScheduler.java

```diff
@@ -134,18 +193,155 @@ private String stageTask(ScheduleRequest scheduleRequest) {
+	 */
 	private Mono<AbstractApplicationSummary> getApplicationByAppName(String appName) {
 		return requestListApplications()
-				.log()
 				.filter(application -> appName.equals(application.getName()))
-				.log()
 				.singleOrEmpty()
 				.cast(AbstractApplicationSummary.class);
 	}
 
+	/**
+	 * Retrieve a  {@link Flux} of {@link ApplicationSummary}s.
+	 */
 	private Flux<ApplicationSummary> requestListApplications() {
 		return this.operations.applications()
 				.list();
 	}
 
+	/**
+	 * Retrieve a cached {@link Flux} of {@link ApplicationSummary}s.
+	 */
+	private Flux<ApplicationSummary> cacheAppSummaries() {
+		return requestListApplications()
+				.cache(); //cache results from first call.  No need to re-retrieve each time.
+	}
+
+	/**
+	 * Retrieve a {@link Flux} containing the available {@link SpaceSummary}s.
+	 * @return {@link Flux} of {@link SpaceSummary}s.
+	 */
+	private Flux<SpaceSummary> requestSpaces() {
+		return this.operations.spaces()
+				.list();
+	}
+
+	/**
+	 * Retrieve a {@link Mono} containing a {@link SpaceSummary} for the specified name.
+	 * @param spaceName the name of space to search.
+	 * @return the {@link SpaceSummary} associated with the spaceName.
+	 */
+	private Mono<SpaceSummary> getSpace(String spaceName) {
+		return requestSpaces()
+				.cache() //cache results from first call.
+				.filter(space -> spaceName.equals(space.getName()))
+				.singleOrEmpty()
+				.cast(SpaceSummary.class);
+	}
+
+	/**
+	 * Retrieve a {@link Mono} containing the {@link ApplicationSummary} associated with the appId.
+	 * @param applicationSummaries {@link Flux} of {@link ApplicationSummary}s to filter.
+	 * @param appId the id of the {@link ApplicationSummary} to search.
+	 */
+	private Mono<ApplicationSummary> getApplication(Flux<ApplicationSummary> applicationSummaries,
+			String appId) {
+		return applicationSummaries
+				.filter(application -> appId.equals(application.getId()))
+				.singleOrEmpty();
+	}
+
+	/**
+	 * Retrieve a Flux of {@link ScheduleInfo}s for the pageNumber specified.
+	 * The PCF-Scheduler returns all data in pages of 50 entries.  This method
+	 * retrieves the specified page and transforms the {@link Flux} of {@link Job}s to
+	 * a {@link Flux} of {@link ScheduleInfo}s
+	 * @param pageNumber integer containing the page offset for the {@link ScheduleInfo}s to retrieve.
+	 * @return {@link Flux} containing the {@link ScheduleInfo}s for the specified page number.
+	 */
+	private Flux<ScheduleInfo> getSchedules(int pageNumber) {
+		Flux<ApplicationSummary> applicationSummaries = cacheAppSummaries();
+		return this.getSpace(this.properties.getSpace()).flatMap(requestSummary -> {
+			return this.client.jobs().list(ListJobsRequest.builder()
+					.spaceId(requestSummary.getId())
+					.page(pageNumber)
+					.detailed(true).build()); })
+				.flatMapIterable
+						(jobs -> jobs.getResources())// iterate over the resources returned.
+				.flatMap(job -> {
```

Comment:
Indentation is odd here.  Why not factor this out into another method?

Reply:
done